include/nfc/ndef/msg_parser.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef NFC_NDEF_MSG_PARSER_H_
+#define NFC_NDEF_MSG_PARSER_H_
+
+/**
+ * @file
+ * @defgroup nfc_ndef_msg_parser Parser for NDEF messages
+ * @{
+ * @brief Parser for NFC NDEF messages and records.
+ */
+
+#include <stdint.h>
+#include <zephyr/types.h>
+#include <nfc/ndef/msg_parser_local.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Macro for calculating the memory size required for holding the
+ *         description of a message that consists of a certain number of
+ *         NDEF records.
+ *
+ *  @param[in] max_count_of_records Maximum number of records to hold.
+ */
+#define NFC_NDEF_PARSER_REQIRED_MEMO_SIZE_CALC(max_count_of_records)            \
+	((u32_t)(max_count_of_records) <= 1) ?                                  \
+	(sizeof(struct nfc_msg_parser_msg_1) * (uint32_t)(max_count_of_records)) : \
+	(sizeof(struct nfc_msg_parser_msg_2) + ((NFC_PARSER_MSG_DELTA) * ((uint32_t)(max_count_of_records) - 1)))
+
+/** @brief Function for parsing NFC NDEF messages.
+ *
+ *  This function parses NDEF messages using NDEF binary record descriptors.
+ *
+ *  @param[out] result_buf Pointer to the buffer that will be used to hold
+ *                         the NDEF message descriptor. After parsing is
+ *                         completed successfully, the first address
+ *                         in the buffer is filled by the NDEF message
+ *                         descriptor (@ref nfc_ndef_msg_desc), which provides
+ *                         a full description of the parsed NDEF message.
+ *  @param[in,out] result_buf_len As input: size of the buffer specified by
+ *                                @p result_buf. As output: size of the reserved
+ *                                (used) part of the buffer specified by
+ *                                @p result_buf.
+ *  @param[in] raw_data Pointer to the data to be parsed.
+ *  @param[in,out] raw_data_len As input: size of the NFC data in
+ *                 the @p nfc_data buffer. As output: size of the
+ *                 parsed message.
+ *
+ *  @retval 0 If the operation was successful.
+ *            Otherwise, a (negative) error code is returned.
+ */
+int nfc_ndef_msg_parser(const u8_t  * const result_buf,
```

Comment:
We need to use an imperative form (verb at the end of function namE)

nfc_ndef_msg_parser ->

nfc_ndef_msg_parse ?

Reply:
Ok